HW9_Task_Management_System_Db_Docker_Jpa/src/main/java/com/hillel/task_management_system/service/TaskServiceFactory.java

```diff
@@ -0,0 +1,29 @@
+package com.hillel.task_management_system.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class TaskServiceFactory {
+
+    @Autowired
+    private TaskServiceJdbc taskServiceJdbc;
+
+    @Autowired
+    private TaskServiceJpa taskServiceJpa;
+
+    @Value("${application.component}")
+    private String componentValue;
+
+    public TaskService createTaskService() {
```

Comment:
Better, use spring approach using @ConditionalOnProperty, will be much less of code